io/solarnet.py

```diff
@@ -0,0 +1,170 @@
+    """
+    return fits.getdata(filename, ext=0)
+
+
+# ========================================================================
+def write(filename, d):
+    """Reads the data as a normal fits (without proper SOLARNET header)
+
+    Arguments:
+        filename: name of the data cube
+        d: data on memory
+
+    Authors: Carlos Diaz (ISP/SU 2019)
+    """
+    io = fits.PrimaryHDU(d)
+    io.writeto(filename, overwrite=True)
+    return
+
+
+# ========================================================================
+def get_wav(filename):
+    """Reads the wavelength information of a 'SOLARNET' cube.
+
+    Arguments:
+        filename: name of the data cube
+        - It is assumed that spectral sampling does not change over time
+
+    Returns:
+        1D wavelength array in nm.
+
+    Example:
+        get_wav(filename, fulltime=False, timeformat=False)
+
+    Authors: Carlos Diaz (ISP/SU 2019)
+    """
+    io = fits.open(filename)
+    # Wavelength information for all wavelength points in the first time frame
+    wav_output = io[1].data[0][0][0,:,0,0,2]
+    return wav_output
+
+
+# ========================================================================
+def seconds2string(number):
+    """Converts time in seconds to 'HH:MM:SS' format
+
+    Arguments:
+        number: value in seconds
+
+    Returns:
+        Time in 'HH:MM:SS' format
+
+    Authors: Carlos Diaz (ISP/SU 2019)
+    """
+    hour = number//3600.
+    minute = (number%3600.)//60.
+    seconds = (number-(hour*3600.+minute*60))/1.
+    string_output = '{0}:{1}:{2:09.6f}'.format(int(hour),int(minute),seconds)
+    return string_output
+
+
+# ========================================================================
+def get_time(filename, fulltime=False, timeformat=False):
+    """Reads the time information of a 'SOLARNET' cube.
+
+    Arguments:
+        filename: name of the data cube
+        fulltime: information at each wavelength point (Default value = False)
+        timeformat: the output is given in 'HH:MM:SS' format (Default value = False)
+
+    Returns:
+        fulltime=False : 1D array with time information at middle wavelength point.
+        fulltime=True : 2D array with time information at each wavelength point.
+
+    Example:
+        data_time = get_time(filename, fulltime=False, timeformat=True)
+
+    Authors: Carlos Diaz (ISP/SU 2019)
+    """
+    io = fits.open(filename)
+
+    wav_len =  io[1].data[0][0].shape[1]
+    middle_wav = wav_len//2
+
+    if fulltime is True:
+        time_output = io[1].data[0][0][:,:,0,0,3]
+
+    else:
+        time_output = io[1].data[0][0][:,middle_wav,0,0,3]
+
+    if timeformat is True:
+        time_output_ = np.array([seconds2string(number) for number in time_output.reshape(time_output.size)])
+        time_output = time_output_.reshape(time_output.shape)
+
+    return time_output
+
+
+# ========================================================================
+def get_extent(filename, timeFrame=0):
```

Comment:
Keyword `timeFrame` is currently not used in the function.

Reply:
Thanks. Solved in last commit.